File: server/src/main/java/com/github/fernthedev/game/server/ServerPacketHandler.java
package com.github.fernthedev.game.server;


import com.github.fernthedev.lightchat.core.packets.Packet;
import com.github.fernthedev.lightchat.server.ClientConnection;
import com.github.fernthedev.lightchat.server.api.IPacketHandler;
import com.github.fernthedev.packets.player_updates.SendPlayerInfoPacket;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class ServerPacketHandler implements IPacketHandler {

    private final GameServer server;

    @Override
    public void handlePacket(Packet p, ClientConnection clientPlayer, int packetId) {
        if (p instanceof SendPlayerInfoPacket) {
            SendPlayerInfoPacket infoPacket = (SendPlayerInfoPacket) p;

            // System.out.println("Received player information " + infoPacket.getPlayerObject());

//            NewServerEntityRegistry serverEntityRegistry = server.getServerGameHandler().getEntityHandler();
//
//            EntityPlayer player = infoPacket.getPlayerObject();

//            serverEntityRegistry.updatePlayerObject(clientPlayer, player);

            if (clientPlayer.getPacketId(p.getClass()).getLeft() -5 < packetId && System.currentTimeMillis() - clientPlayer.getPacketId(p.getClass()).getRight() > 900)
                server.getServerGameHandler().getEntityHandler().handleClientRespond(clientPlayer, infoPacket);

            //UniversalHandler.getThingHandler().updatePlayerObject(infoPacket.getPlayerObject());

            //Server.sendObjectToAllPlayers(infoPacket);
        }
    }
}

File: server/src/main/java/com/github/fernthedev/game/server/ServerPacketHandler.kt
package com.github.fernthedev.game.server

import com.github.fernthedev.lightchat.core.codecs.AcceptablePacketTypes
import com.github.fernthedev.lightchat.core.packets.PacketProto
import com.github.fernthedev.lightchat.server.ClientConnection
import com.github.fernthedev.lightchat.server.api.IPacketHandler
import com.github.fernthedev.packets.proto.Packets


class ServerPacketHandler(
    val server: GameServer
) : IPacketHandler {
    override suspend fun handlePacket(
        acceptablePacketTypes: AcceptablePacketTypes,
        clientConnection: ClientConnection,
        packetId: Int
    ) {
        val proto = acceptablePacketTypes as? PacketProto

        val infoPacket = proto?.message as? Packets.ClientWorldUpdatePacket
        if (infoPacket != null) {
            val packetIdAndTime: Pair<Int, Long> = clientConnection.getPacketId(acceptablePacketTypes.javaClass)
            val id = packetIdAndTime.first
            val time = packetIdAndTime.second

//            if (id -5 < packetId && System.currentTimeMillis() - time > 900)
            server.serverGameHandler.entityHandler.handleClientRespond(clientConnection, infoPacket)
        }
    }
}